### PR TITLE
Lint: prepare `lib/utilities` for `shellcheck`

### PR DESCRIPTION
**Pull request by gaelicWizard**
## Description
This is a partial PR covering only the `lib/utilities` portion of #1917 in support of #1696.

## Motivation and Context
My `shellcheck` branch (#1917) got a bit large, so I'm splitting it up. Each change is split out in to its own commit, so it should be easy to review that way. There are no intended behavior changes whatsoever; this is almost entirely quoting things, plus a few improvements to functions. 

## How Has This Been Tested?
The full test suite and `shellcheck` have run on #1917 and this is part of my main branch.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [x] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

**Comment by gaelicWizard** [comment]
@cornfeedhobo another PR if you're looking for something to read 😆

**Comment by NoahGorny** [comment]
I like this, this requires a rebase, but looks great overall. You can also add some tests to `utilities.bats` when you are at it- that would be amazing :heart:

**Comment by gaelicWizard** [comment]
Rebased to current master and all tests pass, including `shfmt`. I'm working on a branch with a bunch of BATS improvements but that will be later

**Review by NoahGorny — APPROVED** [review]
Great work @gaelicWizard!
Looking forward for a bats PR :smile:

File: lib/utilities.bash [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
-  # filename without path or priority
-  filename=${filename##*---}
-  # filename without path, priority or extension
-  echo ${filename%.*.bash}
+function _bash-it-get-component-name-from-path() {
+	local filename
+	# filename without path
+	filename="${1##*/}"
+	# filename without path or priority
+	filename="${filename##*${BASH_IT_LOAD_PRIORITY_SEPARATOR?}}"
```
Comment (NoahGorny) [inline lib/utilities.bash:14]: seems like this generates SC2295 warning, somehow this slipped here @gaelicWizard

**Pull request merged**